tests/unit/widget/widget_classes.js

```diff
@@ -0,0 +1,137 @@
+(function( $ ) {
+
+module( "widget factory", {
+	teardown: function() {
+		if ( $.ui ) {
+			delete $.ui.testWidget;
+			delete $.fn.testWidget;
+		}
+	}
+});
+
+TestHelpers.testJshint( "widget" );
+
+$.widget( "ui.classesWidget", {
+	options: {
+		classes: {
+			"test-wrapper": "self-wrapper",
+			"test-self": "self-class self-class-2"
+		}
+	},
+	defaultElement: "<div>",
+	_create: function() {
+		this.span = $( "<span>" )
+			.appendTo( this.element );
+
+		this.element.wrap( "<div>" );
+
+		this.wrapper = this.element.parent();
+
+		this._addClass( "test-self", "test-self-extra" )
+			._addClass( "test-self-2" )
+			._addClass( null, "test-self-extra-null" )
+			._addClass( this.span, null, "test-span-extra-null" )
+			._addClass( this.span, "test-span", "test-span-extra" )
+			._addClass( this.wrapper, "test-wrapper" );
+
+	},
+	toggleClasses: function( bool ) {
+		this._toggleClass( "test-self", "test-self-extra", bool )
+			._toggleClass( "test-self-2", null, bool )
+			._toggleClass( null, "test-self-extra-null", bool )
+			._toggleClass( this.span, null, "test-span-extra-null", bool )
+			._toggleClass( this.span, "test-span", "test-span-extra", bool )
+			._toggleClass( this.wrapper, "test-wrapper", null, bool );
+	},
+	removeClasses: function() {
+		this._removeClass( "test-self", "test-self-extra" )
+			._removeClass( "test-self-2" )
+			._removeClass( null, "test-self-extra-null" )
+			._removeClass( this.span, null, "test-span-extra-null" )
+			._removeClass( this.span, "test-span", "test-span-extra" )
+			._removeClass( this.wrapper, "test-wrapper" );
+	},
+	_destroy: function() {
+		this.span.remove();
+		this.element.unwrap();
+	},
+	widget: function() {
+		return this.wrapper;
+	}
+});
+
+test( ".option() - classes setter", function(){
+	expect( 6 );
+
+	var testWidget = $.ui.classesWidget(),
+		currentWrapperClass = testWidget.option( "classes.test-wrapper" );
+
+	testWidget.option({
+		classes: {
+			"test-span": "self-span-new",
+			"test-wrapper": currentWrapperClass + " self-wrapper-new",
+			"test-self": "self-class-2"
+		}
+	});
+
+	equal( testWidget.element.is( ".test-self.self-class-2" ), true,
+		"Removing a class leaves the structure and other classes in value" );
+	equal( !testWidget.element.is( ".self-class" ), true,
+		"Removing a class from the value removes the class" );
```

Comment:
This is actually a place where you should be testing the full `class` attribute.